src/QsCompiler/BondSchemas/CompilerDataStructures.bond

```diff
@@ -0,0 +1,911 @@
+{
+    PauliX,
+    PauliY,
+    PauliZ,
+    PauliI
+}
+
+// N.B. Does not completely represent F#'s QsExpressionKind.
+//      Used by QsExpressionKindComposition to do a complete representation of the original F# type.
+enum QsExpressionKind
+{
+    UnitValue,
+    Identifier,
+    ValueTuple,
+    IntLiteral,
+    BigIntLiteral,
+    DoubleLiteral,
+    BoolLiteral,
+    StringLiteral,
+    ResultLiteral,
+    PauliLiteral,
+    RangeLiteral,
+    NewArray,
+    ValueArray,
+    ArrayItem,
+    NamedItem,
+    NEG,
+    NOT,
+    BNOT,
+    ADD,
+    SUB,
+    MUL,
+    DIV,
+    MOD,
+    POW,
+    EQ,
+    NEQ,
+    LT,
+    LTE,
+    GT,
+    GTE,
+    AND,
+    OR,
+    BOR,
+    BAND,
+    BXOR,
+    LSHIFT,
+    RSHIFT,
+    CONDITIONAL,
+    CopyAndUpdate,
+    UnwrapApplication,
+    AdjointApplication,
+    ControlledApplication,
+    CallLikeExpression,
+    MissingExpr,
+    InvalidExpr
+}
+
+// Used to encapsulate data when QsExpressionKind is Identifier.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindIdentifier<TSymbol, TType>
+{
+    5: required TSymbol Symbol;
+
+    10: nullable<vector<TType>> Types = nothing;
+}
+
+// Used to encapsulate data when QsExpressionKind is StringLiteral.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindStringLiteral<TExpression>
+{
+    5: required string StringLiteral;
+
+    10: required vector<TExpression> Expressions;
+}
+
+// Used to encapsulate data when QsExpressionKind is RangeLiteral, ArrayItem, ADD, SUB, MUL, DIV, MOD, POW, EQ, NEQ, LT, LTE, GT, GTE, AND, OR, BOR, BAND, BXOR, LSHIFT, RSHIFT, CallLikeApplication.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindExpressionDouble<TExpression>
+{
+    5: required TExpression Expression1;
+
+    10: required TExpression Expression2;
+}
+
+// Used to encapsulate data when QsExpressionKind is CONDITIONAL, CopyAndUpdate.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindExpressionTriple<TExpression>
+{
+    5: required TExpression Expression1;
+
+    10: required TExpression Expression2;
+
+    15: required TExpression Expression3;
+}
+
+// Used to encapsulate data when QsExpressionKind is NewArray.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindNewArray<TExpression, TType>
+{
+    5: required TType Type;
+
+    10: required TExpression Expression;
+}
+
+// Used to encapsulate data when QsExpressionKind is NamedItem.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindNamedItem<TExpression, TSymbol>
+{
+    5: required TExpression Expression;
+
+    10: required TSymbol Symbol;
+}
+
+// Completely represents F#'s QsExpressionKind.
+// N.B. Does not have an equivalent F# type.
+struct QsExpressionKindComposition<TExpression, TSymbol, TType>
+{
+    5: required QsExpressionKind Kind = InvalidExpr;
+
+    // Not null when Kind is Identifier.
+    10: nullable<QsExpressionKindIdentifier<TSymbol, TType>> Identifier = nothing;
```

Comment:
`nullable` usually doesn't need to be used along with `nothing`. See [A Thorough Guide to Bond for C#](https://microsoft.github.io/bond/manual/bond_cs.html#default-value-of-nothing) which is actually not all that thorough, but does have some detail on `nullable` and `nothing`. When you specify that a given property is `nullable` it will already default to null when the type for that property is initialized.

Reply:
Thanks for pointing this out! I remember reading that part but at some point during my testing I was seeing that if `nothing` is not explicitly set and the field is omitted when building the object, the field is set to a default instance of the object type rather than null. 

I am not sure if this is because of some early set up that I had but I am not seeing the behavior that I described anymore. Having said that, I will remove the explicit sets to null unless I can reproduce the other behavior.

Reply:
All of the redundant `nothing` is now removed from the schemas.